### PR TITLE
Archiving improvements

### PR DESCRIPTION
**Pull request by jar398**
This is primarily an overhaul of the Makefile to make OTT source updates and OTT releases easier and more automatic.

It says 88 changed files, but most of those are renamings and/or small changes.

User-facing documentation is forthcoming in a separate PR.

*Added 15 May:* New code for importing WoRMS.

### CODE REVIEW

**Comment by kcranston** [comment]
Which files are the key ones to review?

File: assemble_ott.py [hunk 1]
```diff
@@ -14,89 +14,120 @@
...
     # those clades from worms and absorb them before NCBI
-    worms = adjustments.load_worms()
+    worms = load_taxonomy('worms')
+    adjustments.adjust_worms(worms)
     # Malacostraca instead of Decapoda because M. is in the separation taxonomy
     (malacostraca, worms_sans_malacostraca) = split_taxonomy(worms, 'Malacostraca')
     align_and_merge(ott.alignment(malacostraca))
     (cnidaria, worms_sans_cnidaria) = split_taxonomy(worms_sans_malacostraca, 'Cnidaria')
     align_and_merge(ott.alignment(cnidaria))
-    (mollusca, low_priority_worms) = split_taxonomy(worms_sans_cnidaria, 'Mollusca')
```
Comment (kcranston) [inline assemble_ott.py]: why did this get deleted?

**Comment by jar398** [comment]
Sorry about the molluscs! I've done lots of merging and rebasing and so on and the change just got lost, I don't know how. I've reinstated them. Thanks for spotting.

**Comment by jar398** [comment]
The Makefile, and then any of its subroutines (shell & python scripts in bin/ and util/) as needed.

Other than that, most of the changes are either trivial (e.g. file renamings) or uninteresting.

But it may not make much sense without some of the documentation on the maintenance branch. The two probably should be reviewed together, perhaps even combined.

I’m still making last-minute fixes to this branch.

**Pull request merged**